libevmasm/Assembly.cpp

```diff
@@ -694,12 +695,20 @@ LinkerObject const& Assembly::assemble() const
 		}
 	}
 
-	assertThrow(
-		immutableReferencesBySub.empty(),
-		AssemblyException,
-		"Some immutables were read from but never assigned."
-	);
-
+	if (!immutableReferencesBySub.empty())
+	{
+		if (_errorReporter)
+			_errorReporter->fatalCodeGenerationError(
+				0000_error,
```

Comment:
Needs fix

Reply:
I also think this check should have happened at an earlier stage, not when generating target code but during the analysis stage.

Reply:
It is very hard to do this at the analysis stage because it can happen that the optimizer removes code.